### PR TITLE
[DS-3488] REST7: Create Item Converter

### PR DESCRIPTION
**Pull request by terrywbrady**
https://jira.duraspace.org/browse/DS-3488?jql=component%20%3D%20new-REST

@abollini , here is a first attempt at an item converter.  

Questions:
1. Will we be exposing Bundle objects through the API or is bundle simple an attribute of a bitstream?
2. The approach in this PR creates bitstream objects when the item object is created.  What is the proper way to expose the link to the bitstream objects rather than automatically loading them?
3. With my latest commit, I have links appearing for collections, but the serialized JSON is displaying null.  Some guidance on how to properly annotate/code DSpace object links would be appreciated.

### CODE REVIEW

File: dspace-spring-rest/src/main/java/org/dspace/app/rest/converter/ItemConverter.java [hunk 1]
```diff
@@ -22,11 +28,34 @@
...
-		// item.setOwningCollection(collectionConverter.fromModel(obj.getOwningCollection()));
+		item.setInArchive(obj.isArchived());
+		item.setDiscoverable(obj.isDiscoverable());
+		item.setWithdrawn(obj.isWithdrawn());
+		item.setLastModified(obj.getLastModified());
+		try {
+			//item.setTemplateItemOf(collectionConverter.fromModel(obj.getTemplateItemOf()));
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
```
Comment (tdonohue) [inline dspace-spring-rest/src/main/java/org/dspace/app/rest/converter/ItemConverter.java]: I realize this is a WIP. But, we should either clean up empty try/catch statements before merging, OR add a TODO comment as to why these are currently commented out (if they are dependent on some future code). There's two of these in this file.

Comment (mwoodiupui) [inline dspace-spring-rest/src/main/java/org/dspace/app/rest/converter/ItemConverter.java]: BTW NetBeans drops nasty yellow flags on foo.printStackTrace() but approves foo.printStackTrace(System.err) which does the same thing.

However, as this is servlet code, it shouldn't be printing stack traces at all; if needed they should be logged.  [LOG.error("What we think happened", e);]

**Comment by terrywbrady** [comment]
I have added some additional logic to make the link to owningCollection functional.

In the HAL browser, the owningCollection link is functional, but the serialization of owningCollection displays as "null".  Some advice on how to better handle this would be appreciated.

**Comment by terrywbrady** [comment]
Added initial handling for community objects

**Review by abollini — COMMENTED** [review]
I have sent you a PR to fix exclude the linked resources in the item serialization (avoiding null to appear) and adding pagination to the community/collection repository.
Once you accept these changes I will merge this PR

**Comment by tdonohue** [comment]
_NOTE:_ Before merging this, I'd recommend either manually squashing these commits (as there are a lot of small ones, and many with uninformative descriptions), **or** using the "Squash and Merge" button in Github (Click the downward arrow on the Merge  button and select "Squash and Merge". Then press it).

**Comment by terrywbrady** [comment]
@tdonohue , I did a rebase/squash per your suggestion

**Pull request merged**